### PR TITLE
[mlir][LLVMIR] Add sinh/cosh/tanh intrinsic ops

### PR DESCRIPTION
**Pull request by pengmai**
This revision adds hyperbolic trigonometric sinh, cosh, and tanh intrinsic ops.

### CODE REVIEW

**Review by wsmoses — APPROVED** [review]

**Review by Dinistro — COMMENTED** [review]
@gysit do we encourage adding intrinsics to the roundtrip tests are are import and export tests enough?

**Comment by Dinistro** [comment]
> @gysit do we encourage adding intrinsics to the roundtrip tests are are import and export tests enough?

We discussed this offline and concluded that this is fine as is. Most intrinsics have no dedicated roundtrip tests, so no need to add it for these simple ones.

**Review by Dinistro — APPROVED** [review]

File: mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir [hunk 1]
```diff
@@ -112,6 +112,25 @@ llvm.func @cos_test(%arg0: f32, %arg1: vector<8xf32>) {
   llvm.return
 }
 
+// CHECK-LABEL: @hyperbolic_trig_test
+llvm.func @hyperbolic_trig_test(%arg0: f32, %arg1: vector<8xf32>) {
+  // CHECK: call float @llvm.sinh.f32
+  "llvm.intr.sinh"(%arg0) : (f32) -> f32
```
Comment (Dinistro) [inline mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir:118]:
```suggestion
  llvm.intr.sinh(%arg0) : (f32) -> f32
```
Is it necessary that this uses the generic format? The import tests do not seem to do so.

Comment (gysit) [reply mlir/test/Target/LLVMIR/llvmir-intrinsics.mlir:118]: Given we have no roundtrip tests I am in favor of using the pretty printed form here!

**Pull request merged**